cdap-cli/src/main/java/co/cask/cdap/cli/command/app/RestartProgramsCommand.java

```diff
@@ -49,14 +49,14 @@ protected BatchProgram createProgram(ProgramRecord programRecord) {
   @Override
   protected void runBatchCommand(PrintStream printStream, Args<BatchProgram> args) throws Exception {
     printStream.print("Stopping programs...\n");
-    programClient.stop(args.appId.getNamespace(), args.programs);
+    programClient.stop(args.appId.getParent().toId(), args.programs);
```

Comment:
nit: store ``args.appId.getParent().toId()`` in a variable so you don't need to create extra objects on line 59